### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by MoRaad97**
### In this PR

- Add unit tests to your classes:
  1. Book
  2. Person
  3. Student
  4. Teacher
  5. Classroom
  6. Rental
  7. all classes that you created while implementing the Decorator design pattern
 
<hr>

### CODE REVIEW

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Profsain, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project of three contributors but only one partner contributes to the project. 

- [ ] Kindly ask your team partners to make commits to this PR.
- [ ] I also notice you have no test case for `Classroom Class` and `Rental Class`, their files are empty.
- [ ] Please also update your readme file with other contributors' details

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Profsain** [comment]
Thank you for the feedback. We will fix the mistake.

File: spec/decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  context TrimmerDecorator do
+    it 'trim the name, limited to 10 char' do
+      trimmer_class = TrimmerDecorator.new($person)
+      name = trimmer_class.correct_name
+      expect(name).to eq("mohammedra")
+    end
+  end
+end
```
Comment (muneebulrehman) [inline spec/decorator_spec.rb]: When running the tests in this file, it is throwing this ⬇️ error. Kindly fix it and submit it for a re-review.
image

File: spec/people_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    end
+
+    it 'can use service should be always true' do
+      can_use = teacher.can_use_service
+      expect(can_use).to be true
+    end
+  end
+
+
+end
```
Comment (muneebulrehman) [inline spec/people_spec.rb]: You also need to test the functionality of adding rentals. Kindly add this test and submit it for a re-review.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Good follow-up with the previous reviewer.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/people_spec.rb (hunk 2 above)
Comment (MoRaad97) [reply spec/people_spec.rb]: Hi @muneebulrehman 

Thanks for the review.
I think we added rentals tests in the `rental_spec.rb` file, please can you explain The requirements more clearly?

Best, 
Mohammed Raad,

File: spec/decorator_spec.rb (hunk 1 above)
Comment (MoRaad97) [reply spec/decorator_spec.rb]: Hi @muneebulrehman 

Thanks for the review.
All the tests are passing in my terminal
ScreenShot_20220908104830

I will double-check the issue.

Best, 
Mohammed Raad,

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the required changes from the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MoRaad97** [comment]
> Hi team😃,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉
> 
> **Highlights**
> 
> * Well done making the required changes from the previous reviewer
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thanks, Happy Coding 🎉🎉🎉
